### ISSUE TITLE
[BUG] Download via ABS only works partially

### ISSUE THREAD

**Issue by skaramuz** [LeoKlaus/plappa#42]
**Describe the bug**

the download of mp3 or m4b files from abs-server only works partially. after approx. 1/4-1/3 of the total amount the download stops or hangs. restarts and reinstalling the app etc. unfortunately do not bring any improvement.

 - Device: iPhone 14 
 - OS: 17.4.1
 - Server Version: ABS 2.8.1

[9.4.2024, 9:40] [CoreDataHandler] Initializing core data store...
[9.4.2024, 9:40] [CoreDataHandler] CoreData store initialized
[9.4.2024, 9:40] [PlaybackStateHandler] Syncing playback states through server
[9.4.2024, 9:40] [StoreKitHandler] unlocked!
[9.4.2024, 9:40] [AudioBookShelfAPIHandler] Successfully pushed playbackstates
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [CoreDataHandler] Creating new task
[9.4.2024, 9:40] [SoundHandler] Couldn't get current chapter
[9.4.2024, 9:40] [SoundHandler] Couldn't get current chapter
[9.4.2024, 9:41] [SoundHandler] Couldn't get current chapter
[9.4.2024, 9:41] [SoundHandler] Couldn't get current chapter
[9.4.2024, 9:41] [SoundHandler] Couldn't get current chapter
[9.4.2024, 9:41] [AudioBookShelfAPIHandler] Successfully pushed playbackstates

**Comment by rbnhln** [comment]
Got kind of the same error. Tried to download an audio book from abs-sever (version 2.8.1), but can only get the first few MBs (around 50 of 760MB). I tried multiple times and every time it stops at a different position.

Found this in the logs:
[10.4.2024, 10:41] [AudioBookShelfAPIHandler] Successfully pushed playbackstates
[10.4.2024, 10:41] [CoreDataHandler] Creating new task

**Comment by LeoKlaus** [comment]
Thank you guys for the infos!

If a download failed, there would be a message about that in the logs.
Can you make sure that before starting the download all chapters are visible?

Are you sure the download itself fails and it's not the progress indicator that get's stuck (you can manually force the indicators to update by closing and reopening the downloads sheet).

It shouldn't matter, but did your device go to sleep or did you switch apps while the download was running?

**Comment by rbnhln** [comment]
> If a download failed, there would be a message about that in the logs. Can you make sure that before starting the download all chapters are visible?

Yes, all chapters were visible before I started the download.

> Are you sure the download itself fails and it's not the progress indicator that get's stuck (you can manually force the indicators to update by closing and reopening the downloads sheet).

Tried this, but the progress bar was stuck at the exact same value. 

> It shouldn't matter, but did your device go to sleep or did you switch apps while the download was running?

This morning I tried both and both did not work. But I tried it just a minute ago and with keeping the display on during the download process it worked well. But let the screen turn off leads to stop the download.

**Comment by LeoKlaus** [comment]
> This morning I tried both and both did not work. But I tried it just a minute ago and with keeping the display on during the download process it worked well. But let the screen turn off leads to stop the download.

That's weird. I just tried on my physical device (iPhone 11 Pro Max) and couldn't reproduce the issue. Do you have any power saving settings on or similar? Are you downloading via wifi or cellular?

On my instance and devices, downloads seem to work very reliably and the only way I can make them fail is by force closing plappa.

**Comment by rbnhln** [comment]
> > This morning I tried both and both did not work. But I tried it just a minute ago and with keeping the display on during the download process it worked well. But let the screen turn off leads to stop the download.
> 
> That's weird. I just tried on my physical device (iPhone 11 Pro Max) and couldn't reproduce the issue. Do you have any power saving settings on or similar? Are you downloading via wifi or cellular?

I tried both. Cellular and WiFi (I tried even different wifi's). I use an iPhone 12 and do not have any specific power saving options enabled. 

> On my instance and devices, downloads seem to work very reliably and the only way I can make them fail is by force closing plappa.

I tested a bit:

I tried to download a book with around 800 MB in the m4b format. I tried it with using both wifi and cellular:

When I start the download and immediately turn off and on the screen the download is still working (same for switching apps). If I do this multiple times, at some point the download is freezing. But the time or the progress of the download is different any time. 


Some additional informations to my setup:

I use ABS (version 2.8.1) --> Traefik Reverse Proxy --> Cloudflare Proxy --> iPhone 12 (iOS 17.4.1) 

I tried to download the same book with the ABS-App (0.9.73-beta) and turned off the screen and switched apps, but the donwload is still proceeding.

EDIT: Disabled Cloudflare Proxy, but still experiencing the issue.

**Comment by LeoKlaus** [comment]
> When I start the download and immediately turn off and on the screen the download is still working (same for switching apps). If I do this multiple times, at some point the download is freezing. But the time or the progress of the download is different any time.

Thanks for the update! I'v been able to reproduce this now. I'll look into it and let you know once I have more info.

**Comment by boomschtick** [comment]
I can confirm the same thing that rbnhln said above.  If anything happens on my phone (iPhone 14) the download will halt and will not resume.  I can duplicate by switching apps or letting my phone screen time out and turn off.  If I leave the download screen active and tap the screen when it goes dim to prevent it from sleeping, I will be able to get the whole download.

**Comment by LeoKlaus** [comment]
I finally took the time to debug this. Turns out that I've moved back to foreground downloadtasks during early testing as the background sessions were causing some big delays for some reason.

I'll publish an update with a fix in the coming days. It'll be on TestFlight for a week or so before it hits the AppStore, as that update introduces some major changes to way files and chapters are handled and I want to make sure that everything works before releasing it publically.

I'll keep you updated!

**Comment by LeoKlaus** [comment]
I've now released the update to an experimental TestFlight channel, if you feel adventurous, you can try it here (you might lose some of your previous downloads): https://testflight.apple.com/join/oSAekRD5

**Comment by skaramuz** [comment]
Thank you for your great work and quick response! It now works perfectly and as expected. I have tried it with large M4B-singe-file-audiobooks and audiobooks from many individual MP3 files and it runs stable and downloads all files without any problems.

**Comment by mattr7m** [comment]
Hello. I've been experiencing this issue and would like to give the test flight fix a try. Can you invite me to the beta please?

**Comment by LeoKlaus** [comment]
> https://testflight.apple.com/join/oSAekRD5

You can use the link I've posted above to join the TestFlight beta.

**Comment by LeoKlaus** [comment]
> Thank you for your great work and quick response! It now works perfectly and as expected. I have tried it with large M4B-singe-file-audiobooks and audiobooks from many individual MP3 files and it runs stable and downloads all files without any problems.

Glad to hear! Thank you for the update! If you find anything else, let me know!
